### PR TITLE
Fixes #47 Automate deploy in Circle CI

### LINKED ISSUE #47: Implement Circle CI
**Feature description**

Continuous integration on Circle CI

**Is your feature request related to a problem? Please describe.**
Running a complex matrix build blocks other projects

**Describe the solution you'd like**
Run matrix builds on [Circle CI](https://circleci.com/) . Focus on release since Travis will stil be used for day to day development .

**Describe alternatives you've considered**
Travis is not an option . There are all sorts of [Github Apps for CI](https://github.com/marketplace/category/continuous-integration) .

**Additional context**
Related to #44 .

**Possible implementation**

- [x] Translate `.travis.yml` rules into `CircleCI` script format
- [x] Enable deployment (needs build matrix)

### PR DESCRIPTION
**Pull request by stdevHan**
Fixes #47

### CODE REVIEW

File: docker/images/circle/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+ARG QEMU_PLATFORM
+FROM balenalib/${QEMU_PLATFORM}-debian-golang
+# FIXME: If not repeated here build fails.
+# See https://travis-ci.org/simelo/libskycoin/jobs/529481211#L649-L653
+
+ARG GITHUB_OAUTH_TOKEN
+ARG CIRCLE_PROJECT_USERNAME
```
Comment (olemis) [inline docker/images/circle/Dockerfile]: Remove `CIRCLE` prefix . Has to be generic , reusable in many services .

File: docker/images/test-arm/Dockerfile [hunk 2]
```diff
@@ -9,6 +9,7 @@ ADD . $GOPATH/src/github.com/skycoin/libskycoin/
 
 RUN [ "cross-build-start" ]
 
+RUN uname -m
```
Comment (olemis) [inline docker/images/test-arm/Dockerfile]: What for ?

Comment (stdevHan) [reply docker/images/test-arm/Dockerfile]: it was to know the architecture that was working

File: docker/images/circle/Dockerfile (hunk 1 above)
Comment (stdevHan) [reply docker/images/circle/Dockerfile]: Okay, good idea

File: .circleci/config.yml [hunk 3]
```diff
@@ -43,23 +41,32 @@ jobs:
...
+
+  publish-github-release_32:
+    docker:
+      - image: docker:18.06.0-ce-git
+    steps:
+      - run: mkdir -p $GOPATH/src/github.com/ $GOPATH/src/github.com/skycoin
+      - checkout
+      - setup_remote_docker:
+          version: 18.06.0-ce
+      - run: if [[ "$CIRCLE_BRANCH" == "master" ]]; then   docker build  --build-arg CIRCLE_SHA1 --build-arg GITHUB_OAUTH_TOKEN --build-arg CIRCLE_PROJECT_USERNAME --build-arg CIRCLE_PROJECT_REPONAME   --build-arg QEMU_PLATFORM=raspberrypi3 --build-arg VERSION  --file $GOPATH/src/github.com/skycoin/libskycoin/docker/images/circle/Dockerfile  $GOPATH/src/github.com/skycoin/libskycoin -t skydev-deploy ; fi
```
Comment (olemis) [inline .circleci/config.yml]: Run `deploy` on Github tags only

File: docker/images/circle/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+RUN ls -oa $GOPATH/src/github.com/skycoin/libskycoin/
+RUN sh $GOPATH/src/github.com/skycoin/libskycoin/ci-scripts/docker_install_debian.sh
+RUN make -C $GOPATH/src/github.com/skycoin/libskycoin dep
+RUN go get github.com/gz-c/gox
+RUN go get -t ./...
+ENV CGO_ENABLED=1
+ENV VERSION=0.25.1
+ENV ARCH=arm
+ENV OS=Linux
```
Comment (olemis) [inline docker/images/circle/Dockerfile]: use `uname`

File: docker/images/circle/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+RUN [ "cross-build-start" ]
+
+RUN ls -oa $GOPATH/src/github.com/skycoin/libskycoin/
+RUN sh $GOPATH/src/github.com/skycoin/libskycoin/ci-scripts/docker_install_debian.sh
+RUN make -C $GOPATH/src/github.com/skycoin/libskycoin dep
+RUN go get github.com/gz-c/gox
+RUN go get -t ./...
+ENV CGO_ENABLED=1
+ENV VERSION=0.25.1
+ENV ARCH=arm
```
Comment (olemis) [inline docker/images/circle/Dockerfile]: use `uname`

**Pull request merged**